wagtail/admin/ui/tables.py

```diff
@@ -0,0 +1,213 @@
+            self.column = column
+
+        def render_html(self, parent_context):
+            return self.column.render_header_html(parent_context)
+
+    class Cell:
+        # Helper object used for rendering table cells in templates -
+        # behaves as a component (i.e. it has a render_html method) but delegates rendering
+        # to Column.render_cell_html
+        def __init__(self, column, instance):
+            self.column = column
+            self.instance = instance
+
+        def render_html(self, parent_context):
+            return self.column.render_cell_html(self.instance, parent_context)
+
+    header_template_name = "wagtailadmin/tables/column_header.html"
+    cell_template_name = "wagtailadmin/tables/cell.html"
+
+    def __init__(self, name, label=None, accessor=None, classname=None, sort_key=None):
+        self.name = name
+        self.accessor = accessor or name
+        self.label = label or capfirst(name.replace('_', ' '))
+        self.classname = classname
+        self.sort_key = sort_key
+        self.header = Column.Header(self)
+
+    def get_header_context_data(self, parent_context):
+        """
+        Compiles the context dictionary to pass to the header template when rendering the column header
+        """
+        table = parent_context['table']
+        return {
+            'column': self,
+            'table': table,
+            'is_orderable': bool(self.sort_key),
+            'is_ascending': self.sort_key and table.ordering == self.sort_key,
+            'is_descending': self.sort_key and table.ordering == ('-' + self.sort_key),
+            'request': parent_context.get('request'),
+        }
+
+    def render_header_html(self, parent_context):
+        """
+        Renders the column's header
+        """
+        context = self.get_header_context_data(parent_context)
+        return render_to_string(self.header_template_name, context)
+
+    def get_value(self, instance):
+        """
+        Given an instance (i.e. any object containing data), extract the field of data to be
+        displayed in a cell of this column
+        """
+        if callable(self.accessor):
+            return self.accessor(instance)
+        else:
+            return multigetattr(instance, self.accessor)
+
+    def get_cell_context_data(self, instance, parent_context):
+        """
+        Compiles the context dictionary to pass to the cell template when rendering a table cell for
+        the given instance
+        """
+        return {
+            'instance': instance,
+            'column': self,
+            'value': self.get_value(instance),
+            'request': parent_context.get('request'),
+        }
+
+    def render_cell_html(self, instance, parent_context):
+        """
+        Renders a table cell containing data for the given instance
+        """
+        context = self.get_cell_context_data(instance, parent_context)
+        return render_to_string(self.cell_template_name, context)
```

Comment:
This is probably going to run extremely slowly since every cell will load, parse, and render a template (and Django templates are really slow as we've seen with Streamfield!)

Reply:
Good point! We could certainly cache the compiled template on the Column object here.